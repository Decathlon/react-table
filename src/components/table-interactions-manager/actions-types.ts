export const UPDATE_CELL_WIDTH = "UPDATE_CELL_WIDTH";
export type UPDATE_CELL_WIDTH = typeof UPDATE_CELL_WIDTH;

export const UPDATE_ROW_HEIGHT = "UPDATE_ROW_HEIGHT";
export type UPDATE_ROW_HEIGHT = typeof UPDATE_ROW_HEIGHT;

export const UPDATE_HIDDEN_COLUMNS_INDEXES = "UPDATE_HIDDEN_COLUMNS_INDEXES";
export type UPDATE_HIDDEN_COLUMNS_INDEXES = typeof UPDATE_HIDDEN_COLUMNS_INDEXES;

export const UPDATE_FIXED_COLUMNS_INDEXES = "UPDATE_FIXED_COLUMNS_INDEXES";
export type UPDATE_FIXED_COLUMNS_INDEXES = typeof UPDATE_FIXED_COLUMNS_INDEXES;

export const UPDATE_FIXED_ROWS_INDEXES = "UPDATE_FIXED_ROWS_INDEXES";
export type UPDATE_FIXED_ROWS_INDEXES = typeof UPDATE_FIXED_ROWS_INDEXES;

export const UPDATE_COLUMNS_CURSOR = "UPDATE_COLUMNS_CURSOR";
export type UPDATE_COLUMNS_CURSOR = typeof UPDATE_COLUMNS_CURSOR;

export const UPDATE_HIDDEN_ROW_INDEXES = "UPDATE_HIDDEN_ROW_INDEXES";
export type UPDATE_HIDDEN_ROW_INDEXES = typeof UPDATE_HIDDEN_ROW_INDEXES;
