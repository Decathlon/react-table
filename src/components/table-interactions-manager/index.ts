export { default as CellDimensionController } from "./cell-dimensions-controller";
export * from "./cell-dimensions-controller";

export { default as ColumnVisibilityController } from "./column-visibility-controller";
export * from "./column-visibility-controller";

export { default as ColumnIdScrollController } from "./column-id-scroll-controller";
export * from "./column-id-scroll-controller";

export { default as FixedColumnController } from "./fixed-column-controller";
export * from "./fixed-column-controller";

export { default as FixedRowController } from "./fixed-row-controller";
export * from "./fixed-row-controller";

export { default as TableInteractionsManager } from "./table-interactions-manager";
export * from "./table-interactions-manager";

export { default as tableManagerReducer } from "./reducers";
export * from "./reducers";
