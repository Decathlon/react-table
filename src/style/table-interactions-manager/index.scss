@import "column-id-scroll-controller";
@import "column-visibility-controller";
@import "common";
