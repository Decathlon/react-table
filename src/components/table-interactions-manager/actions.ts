import * as actionTypes from "./actions-types";
import { CellValue, CellDimension } from "./reducers";

export interface UpdateCellWidth {
  type: actionTypes.UPDATE_CELL_WIDTH;
  value: CellDimension;
}

export const updateCellWidth = (value: CellDimension): UpdateCellWidth => ({
  type: actionTypes.UPDATE_CELL_WIDTH,
  value
});

export interface UpdateRowHeight {
  type: actionTypes.UPDATE_ROW_HEIGHT;
  value: CellDimension;
}

export const updateRowHeight = (value: CellDimension): UpdateRowHeight => ({
  type: actionTypes.UPDATE_ROW_HEIGHT,
  value
});

export interface UpdateHiddenColumns {
  type: actionTypes.UPDATE_HIDDEN_COLUMNS_INDEXES;
  hiddenColumnsIds: string[];
}

export const updateHiddenColumns = (hiddenColumnsIds: string[]): UpdateHiddenColumns => ({
  type: actionTypes.UPDATE_HIDDEN_COLUMNS_INDEXES,
  hiddenColumnsIds
});

export interface UpdateHiddenRows {
  type: actionTypes.UPDATE_HIDDEN_ROW_INDEXES;
  hiddenRowIndexes: number[];
}

export const updateHiddenRows = (hiddenRowIndexes: number[]): UpdateHiddenRows => ({
  type: actionTypes.UPDATE_HIDDEN_ROW_INDEXES,
  hiddenRowIndexes
});

export interface UpdateFixedColumns {
  type: actionTypes.UPDATE_FIXED_COLUMNS_INDEXES;
  fixedColumnsIds: string[];
}

export const updateFixedColumns = (fixedColumnsIds: string[]): UpdateFixedColumns => ({
  type: actionTypes.UPDATE_FIXED_COLUMNS_INDEXES,
  fixedColumnsIds
});

export interface UpdateFixedRows {
  type: actionTypes.UPDATE_FIXED_ROWS_INDEXES;
  fixedRowsIndexes: number[];
}

export const updateFixedRows = (fixedRowsIndexes: number[]): UpdateFixedRows => ({
  type: actionTypes.UPDATE_FIXED_ROWS_INDEXES,
  fixedRowsIndexes
});

export interface UpdateColumnsCursor {
  type: actionTypes.UPDATE_COLUMNS_CURSOR;
  columnsCursor: CellValue;
}

export const updateColumnsCursor = (columnsCursor: CellValue): UpdateColumnsCursor => ({
  type: actionTypes.UPDATE_COLUMNS_CURSOR,
  columnsCursor
});

export type TableInteractionsAction =
  | UpdateCellWidth
  | UpdateRowHeight
  | UpdateHiddenColumns
  | UpdateHiddenRows
  | UpdateColumnsCursor
  | UpdateFixedColumns
  | UpdateFixedRows;
