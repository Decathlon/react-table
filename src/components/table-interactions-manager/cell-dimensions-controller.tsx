import * as React from "react";
import Menu from "@material-ui/core/Menu";
import Divider from "@material-ui/core/Divider";
import ListSubheader from "@material-ui/core/ListSubheader";
import { Icon, ListItemText, List, MenuItem, ListItemIcon } from "@material-ui/core";

import { TableInteractionsContext } from "./table-interactions-manager";
import { CellSize, CELL_SIZES, CellDimention } from "./reducers";
import { Nullable } from "../typing";

export interface ICellDimensionControllerProps {
  /** The current cell width of the table. */
  cellWidth: CellDimention;
  /** The current row Height of the table. */
  rowHeight: CellDimention;
  /** The menu button activator renderer */
  buttonRenderer: (toggleMenu: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void) => JSX.Element;
  /** The row height controler. Please see the CellDimensionController */
  updateRowHeight: (size: CellSize) => void;
  /** The cell width controler. Please see the CellDimensionController */
  updateCellWidth: (size: CellSize) => void;
}

export const DumbCellDimensionController: React.FunctionComponent<ICellDimensionControllerProps> = React.memo(
  ({ updateCellWidth, updateRowHeight, buttonRenderer, rowHeight, cellWidth }) => {
    const [anchorEl, setAnchorEl] = React.useState<Nullable<Element>>(null);
    const isOpen = Boolean(anchorEl);

    const onClose = () => {
      setAnchorEl(null);
    };

    const toggleMenu = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
      setAnchorEl(anchorEl ? null : event.currentTarget);
    };

    const getColumnWidthUpdater = (size: CellSize) => () => {
      updateCellWidth(size);
    };

    const getRowHeightUpdater = (size: CellSize) => () => {
      updateRowHeight(size);
    };
    return (
      <>
        {buttonRenderer(toggleMenu)}
        <Menu
          data-testid="cell-dimensions-menu"
          elevation={2}
          open={isOpen}
          anchorOrigin={{ horizontal: "center", vertical: "bottom" }}
          transformOrigin={{ horizontal: "right", vertical: "top" }}
          anchorEl={anchorEl}
          onClose={onClose}
        >
          <List classes={{ root: "table-interaction-menu" }}>
            <ListSubheader>Columns</ListSubheader>
            {CELL_SIZES.map(size => (
              <MenuItem
                key={`column-${size}`}
                data-testid={`column-width-dimension-${size}`}
                onClick={getColumnWidthUpdater(size)}
              >
                <ListItemIcon>
                  {cellWidth.size === size ? (
                    <Icon data-testid="column-width-dimension-checked">radio_button_checked</Icon>
                  ) : (
                    <Icon data-testid="column-width-dimension-unchecked">radio_button_unchecked</Icon>
                  )}
                </ListItemIcon>

                <ListItemText id={size} primary={size} />
              </MenuItem>
            ))}
            <Divider />
            <ListSubheader>Rows</ListSubheader>
            {CELL_SIZES.map(size => (
              <MenuItem key={`row-${size}`} data-testid={`row-height-dimension-${size}`} onClick={getRowHeightUpdater(size)}>
                <ListItemIcon>
                  {rowHeight.size === size ? (
                    <Icon data-testid="row-height-dimension-checked">radio_button_checked</Icon>
                  ) : (
                    <Icon data-testid="row-height-dimension-unchecked">radio_button_unchecked</Icon>
                  )}
                </ListItemIcon>
                <ListItemText id={size} primary={size} />
              </MenuItem>
            ))}
          </List>
        </Menu>
      </>
    );
  }
);

const CellDimensionController: React.FunctionComponent<Pick<ICellDimensionControllerProps, "buttonRenderer">> = props => {
  return (
    <TableInteractionsContext.Consumer>
      {({ cellWidth, rowHeight, updateCellWidth, updateRowHeight }) => {
        return (
          <DumbCellDimensionController
            cellWidth={cellWidth}
            rowHeight={rowHeight}
            updateCellWidth={updateCellWidth}
            updateRowHeight={updateRowHeight}
            {...props}
          />
        );
      }}
    </TableInteractionsContext.Consumer>
  );
};

export default React.memo(CellDimensionController);
