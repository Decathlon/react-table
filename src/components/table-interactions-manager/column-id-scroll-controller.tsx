import * as React from "react";
import { IconButton, Icon } from "@mui/material";
import Select from "@mui/material/Select";
import MenuItem from "@mui/material/MenuItem";

import { TableInteractionsContext } from "./table-interactions-manager";

export interface IColumnLabel {
  id: string;
  label: string;
}

export interface IColumnIdScrollControllerProps {
  /** Scrollable columns to control */
  columns: IColumnLabel[];
}

export interface IDumbIColumnIdScrollControllerPropsProps extends IColumnIdScrollControllerProps {
  /** The current columns cursor of the table (current scroll). */
  columnsCursorId?: string;
  /** The scroll controler (scrolling by column id). Please see the WeekScrollerController. */
  goToColumnId: (columnId: string) => void;
}

export const DumbColumnIdScrollController: React.FunctionComponent<IDumbIColumnIdScrollControllerPropsProps> = ({
  columns,
  columnsCursorId,
  goToColumnId,
}) => {
  const selectedColumnIndex = React.useMemo(
    () => columns.findIndex((column) => column.id === columnsCursorId),
    [columnsCursorId]
  );

  const gotToColumn = (columnIndex: number) => {
    const column = columns[columnIndex];
    if (column) {
      goToColumnId(column.id);
    }
  };

  const onChange = (event: React.ChangeEvent<{ name?: string; value: number }>) => {
    gotToColumn(event.target.value);
  };

  return (
    <div className="scroll-controlller">
      <IconButton
        id="table-toolbar_prev-columns"
        className="scroll-controlller__button"
        onClick={() => gotToColumn(0)}
        size="large"
      >
        <Icon>first_page</Icon>
      </IconButton>

      <IconButton
        id="table-toolbar_prev-column"
        className="scroll-controlller__button"
        onClick={() => gotToColumn(selectedColumnIndex - 1)}
        size="large"
      >
        <Icon>chevron_left</Icon>
      </IconButton>

      <Select
        MenuProps={{ id: "column-id-scroll-controller" }}
        autoWidth
        id="table-toolbar_current-column"
        className="scroll-controlller__field"
        value={selectedColumnIndex}
        // @ts-ignore value is a number
        onChange={onChange}
        variant="standard"
      >
        {columns.map((column, index) => (
          <MenuItem key={`column_${column.id}`} id={`column-scroll-item_${column.id}`} value={index}>
            {column.label}
          </MenuItem>
        ))}
      </Select>

      <IconButton
        id="table-toolbar_next-column"
        className="scroll-controlller__button"
        onClick={() => gotToColumn(selectedColumnIndex + 1)}
        size="large"
      >
        <Icon>chevron_right</Icon>
      </IconButton>

      <IconButton
        id="table-toolbar_next-columns"
        className="scroll-controlller__button"
        onClick={() => gotToColumn(columns.length - 1)}
        size="large"
      >
        <Icon>last_page</Icon>
      </IconButton>
    </div>
  );
};

const ColumnIdScrollController: React.FunctionComponent<IColumnIdScrollControllerProps> = (props) => {
  return (
    <TableInteractionsContext.Consumer>
      {({ columnsCursor, goToColumnId }) => {
        return (
          <DumbColumnIdScrollController
            goToColumnId={goToColumnId}
            columnsCursorId={columnsCursor ? columnsCursor.id : undefined}
            {...props}
          />
        );
      }}
    </TableInteractionsContext.Consumer>
  );
};

export default React.memo(ColumnIdScrollController);
