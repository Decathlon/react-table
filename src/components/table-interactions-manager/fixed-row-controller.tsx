import * as React from "react";

import { TableInteractionsContext } from "./table-interactions-manager";

interface IFixedRowChildrenProps {
  toggleFixedRowIndex: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
  isFixed: boolean;
}

interface IFixedRowControllerProps {
  rowIndex: number;
  children: (props: IFixedRowChildrenProps) => JSX.Element;
}

interface IDumbFixedRowControllerProps extends IFixedRowControllerProps {
  /** The current fixed Rows of the table (indexes). */
  fixedRowsIndexes: number[];
  /** The fixed Rows controller. Please see the FixedRowController. */
  updateFixedRowsIndexes: (fixedRowsIndexes: number[]) => void;
}

export const DumbFixedRowController: React.FunctionComponent<IDumbFixedRowControllerProps> = React.memo(
  ({ children, rowIndex, updateFixedRowsIndexes, fixedRowsIndexes }) => {
    const isFixed = React.useMemo(() => {
      return fixedRowsIndexes.includes(rowIndex);
    }, [rowIndex, fixedRowsIndexes]);
    const toggleFixedRowIndex = () => {
      const newRows = fixedRowsIndexes.filter(fixedRowIndex => fixedRowIndex !== rowIndex);
      if (newRows.length === fixedRowsIndexes.length) {
        newRows.push(rowIndex);
      }
      updateFixedRowsIndexes(newRows);
    };

    return children({ toggleFixedRowIndex, isFixed });
  }
);

const FixedRowController: React.FunctionComponent<IFixedRowControllerProps> = props => {
  const { fixedRowsIndexes, updateFixedRowsIndexes } = React.useContext(TableInteractionsContext);
  return (
    <DumbFixedRowController fixedRowsIndexes={fixedRowsIndexes} updateFixedRowsIndexes={updateFixedRowsIndexes} {...props} />
  );
};

export default React.memo(FixedRowController);
