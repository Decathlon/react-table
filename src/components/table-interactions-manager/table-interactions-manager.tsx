import * as React from "react";

import Table from "../table/table";
import { OnHorizontallyScrollProps } from "../virtualizer";
import { TableInteractionsAction, updateHiddenColumns, updateRowHeight, updateCellWidth, updateColumnsCursor } from "./actions";
import TableInteractionsManagerReducer, { IState, CellSize, CellValue, initialState } from "./reducers";
import { ICellCoordinates, ICell } from "../table/cell";
import { Nullable } from "../typing";

export interface OnScrollCallbackProps {
  /** The current column id. */
  columnsCursorId: string;
}

interface Column {
  id: string;
  index: number;
}

interface IProvider extends IState {
  /** The table ref. */
  table: Nullable<React.MutableRefObject<Nullable<Table>>>;
  /** The current hidden columns of the table (indexes). */
  hiddenColumnsIndexes: number[];
  /** The hidden columns controller. Please see the ColumnVisibilityController. */
  updateHiddenIds: (hiddenIds: string[]) => void;
  /** The row height controler. Please see the CellDimensionController */
  updateRowHeight: (size: CellSize) => void;
  /** The cell width controler. Please see the CellDimensionController */
  updateCellWidth: (size: CellSize) => void;
  /** The scroll controler (scrolling by column id). Please see the WeekScrollerController. */
  goToColumnId: (columnId: string) => void;
  /** The scroll controler (scrolling by column index). Please see the WeekScrollerController. */
  goToColumnIndex: (columnIndex: number) => void;
  /** return the cell props of the table */
  getCell: (cellCoordinates: ICellCoordinates) => Nullable<ICell>;
  /** Callback fired when a "scroll" (Horizontally) event is detected. */
  onHorizontallyScroll: (
    props: OnHorizontallyScrollProps,
    callback?: (onScrollCallbackProps: OnScrollCallbackProps) => void
  ) => void;
}

interface IProps {
  children: JSX.Element | JSX.Element[];
  toggleableColumns: Column[];
  /** The initial table interaction config. */
  initialConfig?: Partial<IState>;
  /** Callback fired when the interaction context state hase changed. */
  onStateUpdate?: (state: IState) => void;
}

const nullFunction = (): any => null;

const initialContentxt: IProvider = {
  ...initialState,
  table: null,
  hiddenColumnsIndexes: [],
  updateHiddenIds: nullFunction,
  updateRowHeight: nullFunction,
  updateCellWidth: nullFunction,
  goToColumnId: nullFunction,
  goToColumnIndex: nullFunction,
  getCell: nullFunction,
  onHorizontallyScroll: nullFunction
};

export const TableInteractionsContext: React.Context<IProvider> = React.createContext<IProvider>({
  ...initialContentxt
});

const TableInteractionsManager = ({ children, initialConfig, onStateUpdate, toggleableColumns = [] }: IProps) => {
  const initalHiddenColumnsIds = React.useMemo(() => (toggleableColumns ? toggleableColumns.map(column => column.id) : []), [
    toggleableColumns
  ]);
  const [state, dispatch] = React.useReducer(TableInteractionsManagerReducer, {
    ...initialState,
    hiddenColumnsIds: initalHiddenColumnsIds,
    ...initialConfig
  });
  const table = React.useRef<Table>(null);
  const { columnsCursor, hiddenColumnsIds } = state;
  const { id: currentColumnsCursorId, index: currentColumnsCursorIndex } = columnsCursor || {
    id: null,
    index: null
  };
  const hiddenColumnsIdsMapping = React.useMemo(
    () =>
      toggleableColumns.reduce((mapping, column) => {
        mapping[column.id] = column.index;
        return mapping;
      }, {}),
    [toggleableColumns]
  );
  const actions = React.useMemo(() => mapDispatchToProps(dispatch), [dispatch]);

  React.useEffect(() => {
    if (onStateUpdate) {
      onStateUpdate(state);
    }
  }, [state]);

  const goToColumnId = React.useCallback(
    (columnId: string) => {
      if (table.current) {
        const columnIndex = table.current.getColumnIndex(columnId);
        actions.updateColumnsCursor({ index: columnIndex, id: columnId });
        table.current.goToColumnId(columnId);
      }
    },
    [actions, table]
  );

  const goToColumnIndex = React.useCallback(
    (columnIndex: number) => {
      if (table.current) {
        const columnId = table.current.getColumnId(columnIndex);
        actions.updateColumnsCursor({ index: columnIndex, id: columnId });
        table.current.goToColumnIndex(columnIndex);
      }
    },
    [actions, table]
  );

  const getCell = React.useCallback(
    (cellCoordinates: ICellCoordinates) => {
      if (table.current) {
        return table.current.getCell(cellCoordinates);
      }
      return null;
    },
    [actions, table]
  );

  const onHorizontallyScroll = React.useCallback(
    (onScrollProps: OnHorizontallyScrollProps, callback?: (onScrollCallbackProps: OnScrollCallbackProps) => void) => {
      const { columnsCursor } = onScrollProps;
      if (table.current) {
        const columnId = table.current.getColumnId(columnsCursor);
        if (table.current && (!currentColumnsCursorIndex || currentColumnsCursorIndex !== columnsCursor)) {
          actions.updateColumnsCursor({ index: columnsCursor, id: columnId });
        }
        if (callback) {
          callback({ columnsCursorId: columnId });
        }
      }
    },
    [actions, table, currentColumnsCursorIndex]
  );

  const updateCellWidth = React.useCallback(
    (size: CellSize) => {
      actions.updateCellWidth(size);
      /** We need an async scrolling. Waiting for cell width update. */
      if (currentColumnsCursorId) {
        setImmediate(() => {
          goToColumnId(currentColumnsCursorId);
        });
      }
    },
    [actions, currentColumnsCursorId, goToColumnId]
  );

  const hiddenColumnsIndexes = React.useMemo(
    () =>
      hiddenColumnsIds.reduce<number[]>((result, columnId) => {
        const columnIndex = hiddenColumnsIdsMapping[columnId];
        if (columnIndex >= 0) {
          result.push(columnIndex);
        }
        return result;
      }, []),
    [hiddenColumnsIds, hiddenColumnsIdsMapping]
  );

  return (
    <TableInteractionsContext.Provider
      value={{
        ...actions,
        ...state,
        hiddenColumnsIndexes,
        updateCellWidth,
        onHorizontallyScroll,
        goToColumnIndex,
        goToColumnId,
        getCell,
        table
      }}
    >
      {children}
    </TableInteractionsContext.Provider>
  );
};

TableInteractionsManager.defaultProps = {
  toggleableColumns: []
};

const mapDispatchToProps = (dispatch: React.Dispatch<TableInteractionsAction>) => ({
  updateHiddenIds: (hiddenIds: string[]) => dispatch(updateHiddenColumns(hiddenIds)),
  updateRowHeight: (size: CellSize) => dispatch(updateRowHeight(size)),
  updateCellWidth: (size: CellSize) => dispatch(updateCellWidth(size)),
  updateColumnsCursor: (columnsCursor: CellValue) => dispatch(updateColumnsCursor(columnsCursor))
});

export default TableInteractionsManager;
