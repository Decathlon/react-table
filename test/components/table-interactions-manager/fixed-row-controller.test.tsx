/// <reference path="../../typings/tests-entry.d.ts" />

import { createRenderer } from "react-test-renderer/shallow";

import { DumbFixedRowController } from "../../../src/components/table-interactions-manager/fixed-row-controller";

describe("FixedRowController component", () => {
  test("should render the default FixedRowController", () => {
    const props = {
      rowIndex: 2,
      fixedRowsIndexes: [1],
      updateFixedRowsIndexes: jest.fn()
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(
      <DumbFixedRowController {...props}>
        {({ toggleFixedRowIndex }) => <span onClick={toggleFixedRowIndex}>Bar</span>}
      </DumbFixedRowController>
    );
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});
