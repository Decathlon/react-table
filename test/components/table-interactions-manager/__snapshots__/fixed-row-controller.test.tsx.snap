// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedRowController component should render the default FixedRowController 1`] = `
<span
  onClick={[Function]}
>
  Bar
</span>
`;
